include/deal.II/physics/notation.h

```diff
@@ -0,0 +1,619 @@
+     * }
+     * @endcode
+     * as well as the online references found <a
+     * href="http://sepwww.stanford.edu/oldsep/joe/Reprints/8IWSA.pdf">here</a> and <a
+     * href="http://books.google.de/books?id=Ptxu63ARr7UC&pg=PA298&lpg=PA298&dq=kelvin+tensor+notation&source=bl&ots=LgKytTBYu9&sig=LHB3IrBRaJLfI8-IEDsBOQz_a4c&hl=en&sa=X&ei=1xZtUeSpBoOqtAa224DQAQ&redir_esc=y#v=onepage&q=kelvin%20tensor%20notation&f=false">here</a>.
+     *
+     * @author Jean-Paul Pelteret, 2017
+     */
+    template <int dim>
+    struct Kelvin
+    {
+      /**
+       * @name Forward operation: Tensor notation to Kelvin notation
+       */
+//@{
+
+      /**
+       * Convert a scalar value to its compressed vector equivalent.
+       */
+      template<typename Number>
+      static
+      Vector<Number>
+      unroll_to_vector (const Number &s);
+
+
+      /**
+       * Convert a rank-0 tensor to its compressed vector equivalent.
+       */
+      template<typename Number>
+      static
+      Vector<Number>
+      unroll_to_vector (const Tensor<0,dim,Number> &s);
+
+
+      /**
+       * Convert a rank-1 tensor to its compressed vector equivalent.
+       */
+      template<typename Number>
+      static
+      Vector<Number>
+      unroll_to_vector (const Tensor<1,dim,Number> &v);
+
+
+      /**
+       * Convert a rank-2 tensor to its compressed vector equivalent.
+       */
+      template<typename Number>
+      static
+      Vector<Number>
+      unroll_to_vector (const Tensor<2,dim,Number> &t);
+
+
+      /**
+       * Convert a rank-2 symmetric tensor to its compressed vector equivalent.
+       */
+      template<typename Number>
+      static
+      Vector<Number>
+      unroll_to_vector (const SymmetricTensor<2,dim,Number> &st);
+
+
+      /**
+       * Convert a scalar value tensor to its compressed matrix equivalent.
+       */
+      template<typename Number>
+      static
+      FullMatrix<Number>
+      unroll_to_matrix (const Number &s);
+
+
+      /**
+       * Convert a rank-0 tensor to its compressed matrix equivalent.
+       */
+      template<typename Number>
+      static
+      FullMatrix<Number>
+      unroll_to_matrix (const Tensor<0,dim,Number> &s);
+
+
+      /**
+       * Convert a rank-0 tensor to its compressed matrix equivalent.
+       */
+      template<typename Number>
+      static
+      FullMatrix<Number>
+      unroll_to_matrix (const Tensor<1,dim,Number> &v);
+
+
+      /**
+       * Convert a rank-2 tensor to its compressed matrix equivalent.
+       */
+      template<typename Number>
+      static
+      FullMatrix<Number>
+      unroll_to_matrix (const Tensor<2,dim,Number> &t);
+
+
+      /**
+       * Convert a rank-2 symmetric tensor to its compressed matrix equivalent.
+       */
+      template<typename Number>
+      static
+      FullMatrix<Number>
+      unroll_to_matrix (const SymmetricTensor<2,dim,Number> &st);
+
+      /**
+       * The template arguments @p SubTensor1 and @p SubTensor2 determine how
```

Comment:
maybe start with what this function does and then what templates mean

Reply:
Yes, that makes sense :-)